luigi/contrib/bigquery.py

```diff
@@ -325,15 +327,25 @@ def run_job(self, project_id, body, dataset=None):
         new_job = self.client.jobs().insert(projectId=project_id, body=body).execute()
         job_id = new_job['jobReference']['jobId']
         logger.info('Started import job %s:%s', project_id, job_id)
+        poll_counter = 0
         while True:
-            status = self.client.jobs().get(projectId=project_id, jobId=job_id).execute()
-            if status['status']['state'] == 'DONE':
-                if status['status'].get('errors'):
-                    raise Exception('BigQuery job failed: {}'.format(status['status']['errors']))
-                return
-
-            logger.info('Waiting for job %s:%s to complete...', project_id, job_id)
-            time.sleep(5.0)
+            try:
+                status = self.client.jobs().get(projectId=project_id, jobId=job_id).execute()
+                if status['status']['state'] == 'DONE':
+                    if status['status'].get('errors'):
+                        raise Exception('BigQuery job failed: {}'.format(status['status']['errors']))
+                    return
+
+                logger.info('Waiting for job %s:%s to complete...', project_id, job_id)
+                poll_counter = 0
+                time.sleep(5.0)
+            except HttpError as httpError:
+                if int(httpError.resp.status) == 503 and poll_counter < self.MAX_POLLING_RETRIES:
+                    poll_counter = poll_counter + 1
+                    logger.warning('Got error 503 while requesting info for job %s:%s, retrying in %d seconds...', project_id, job_id, poll_counter)
+                    time.sleep(float(poll_counter))
```

Comment:
Tt doesn't have to be float, at least according to docstring of `time.sleep`. You could remove the `.0` in the other call too.